src/server/api/routers/s3.ts

```
@@ -66,5 +67,19 @@ export const s3Router = createTRPCRouter({
         console.error("Error deleting file:", error);
         throw new Error("Failed to delete file.");
       }
+    }),
+  downloadFile: publicProcedure
+    .input(z.object({ fileKey: z.string() }))
+    .mutation(async ({ input }) => {
+      const file = await getFileFromS3(input.fileKey);
+      if (!file.Body) {
+        throw new Error("File not found");
+      }
+      return {
+        content:
+          file.Body instanceof Buffer ? file.Body.toString("base64") : "",
```

we shouldn't return `""` as a content, this would cause an empty download, 

let's instead return an error in that case:

```typescript
downloadFile: publicProcedure
  .input(z.object({ fileKey: z.string() }))
  .mutation(async ({ input }) => {
    const file = await getFileFromS3(input.fileKey);
    if (!file || !file.Body || !(file.Body instanceof Buffer)) {
      throw new Error("File not found or invalid file type");
    }
    return {
      content: file.Body.toString("base64"),
      contentType: file.ContentType,
      fileName: input.fileKey.split("/").pop() ?? "download"
    };
  })


```

---

I think this is missing?